src/WebJobs.Extensions.DurableTask/OutOfProcMiddleware.cs

```diff
@@ -145,6 +145,12 @@ public async Task CallOrchestratorAsync(DispatchMiddlewareContext dispatchContex
                 functionResult = await function.Executor.TryExecuteAsync(
                     input,
                     cancellationToken: this.HostLifetimeService.OnStopping);
+                if (!functionResult.Succeeded)
+                {
+                    // Shutdown can surface as a completed invocation in a failed state.
+                    // Re-throw so we can abort this invocation.
+                    this.HostLifetimeService.OnStopping.ThrowIfCancellationRequested();
```

Comment:
I assume that this throws `OperationCanceledException`, and that our existing logic knows how to gracefully abort when this exception is encountered?

Reply:
Correct - this is inside a `try/catch` block, which will catch that exception and rethrow a `SessionAbortedException`